packages/firestore/exp/src/api/reference.ts

```diff
@@ -83,6 +84,63 @@ export function getDocFromServer<T>(
   });
 }
 
+export function getQuery<T>(
+  query: firestore.Query<T>
+): Promise<QuerySnapshot<T>> {
+  const internalQuery = cast<Query<T>>(query, Query);
+  const firestore = cast<Firestore>(query.firestore, Firestore);
+  return firestore._getFirestoreClient().then(async firestoreClient => {
```

Comment:
Can you make methods like this fully async? I'm not seeing why they need mixed notation.  This applies to the other functions below.

Reply:
The outermost code needs to be async to ensure synchronous function execution, which impacts error propagation and execution order. The `cast` calls here, for example, would not throw but cause rejected Promises.